models/DataObject/ClassDefinition/LinkGeneratorInterface.php

```diff
@@ -15,15 +15,16 @@
 
 namespace Pimcore\Model\DataObject\ClassDefinition;
 
+use Pimcore\Bundle\EcommerceFrameworkBundle\Model\DefaultMockup;
 use Pimcore\Model\DataObject\Concrete;
 
 interface LinkGeneratorInterface
 {
     /**
-     * @param Concrete $object
+     * @param Concrete|DefaultMockup $object
      * @param array $params
      *
      * @return string
      */
-    public function generate(Concrete $object, array $params = []): string;
+    public function generate(Concrete|DefaultMockup $object, array $params = []): string;
```

Comment:
Should the `EcommerceFrameworkBundle` code be mixed into the general non-bundle code?

Reply:
Good point - maybe just allowing any object is sufficient because in the end it depends on the generator how to handle it....
Changed it to 
`
public function generate(object $object, array $params = []): string;
`